src/apm-cli.coffee

```diff
@@ -100,34 +100,50 @@ printVersions = (args, callback) ->
 
   getPythonVersion (pythonVersion) ->
     git.getGitVersion (gitVersion) ->
-      if args.json
-        versions =
-          apm: apmVersion
-          npm: npmVersion
-          node: nodeVersion
-          python: pythonVersion
-          git: gitVersion
-          nodeArch: process.arch
-        if config.isWin32()
-          versions.visualStudio = config.getInstalledVisualStudioFlag()
-        console.log JSON.stringify(versions)
-      else
-        pythonVersion ?= ''
-        gitVersion ?= ''
-        versions =  """
-          #{'apm'.red}  #{apmVersion.red}
-          #{'npm'.green}  #{npmVersion.green}
-          #{'node'.blue} #{nodeVersion.blue} #{process.arch.blue}
-          #{'python'.yellow} #{pythonVersion.yellow}
-          #{'git'.magenta} #{gitVersion.magenta}
-        """
-
-        if config.isWin32()
-          visualStudioVersion = config.getInstalledVisualStudioFlag() ? ''
-          versions += "\n#{'visual studio'.cyan} #{visualStudioVersion.cyan}"
-
-        console.log versions
-      callback()
+      getAtomVersion (atomVersion) ->
+        if args.json
+          versions =
+            apm: apmVersion
+            npm: npmVersion
+            node: nodeVersion
+            atomVersion: atomVersion
+            python: pythonVersion
+            git: gitVersion
+            nodeArch: process.arch
+          if config.isWin32()
+            versions.visualStudio = config.getInstalledVisualStudioFlag()
+          console.log JSON.stringify(versions)
+        else
+          pythonVersion ?= ''
+          gitVersion ?= ''
+          atomVersion ?= ''
+          versions =  """
+            #{'apm'.red}  #{apmVersion.red}
+            #{'npm'.green}  #{npmVersion.green}
+            #{'node'.blue} #{nodeVersion.blue} #{process.arch.blue}
+            #{'atom'.cyan} #{atomVersion.cyan}
+            #{'python'.yellow} #{pythonVersion.yellow}
+            #{'git'.magenta} #{gitVersion.magenta}
+          """
+
+          if config.isWin32()
+            visualStudioVersion = config.getInstalledVisualStudioFlag() ? ''
+            versions += "\n#{'visual studio'.cyan} #{visualStudioVersion.cyan}"
+
+          console.log versions
+        callback()
+
+getAtomVersion = (callback) ->
+  config.getResourcePath (resourcePath) ->
+    unknownVersion = 'unknown'
+    # workaround if app.asar does not exist. see: https://github.com/atom/atom/issues/6604
```

Comment:
We haven't seen any reports of 6604 in a while.  Is this needed?  Did you run into it?

Reply:
Yes. https://github.com/atom/apm/pull/745#discussion_r141298972

Reply:
Whoops!  Totally forgot I commented about it before 😊

Reply:
Atom 1.21 added back `app.asar`, see: https://github.com/atom/apm/pull/738#issuecomment-337573518

For users who have Atom 1.20 and any other versions which do not use `app.asar` the workaround check seems to be needed.

Reply:
Ok, good to know.  Since apm is bundled with Atom and can't be installed manually, I think it's safe to remove this check since it'll be bundled with at least Atom 1.23 at this point.

Reply:
Done. See e3970212f52a100a8467a1e844e6a148808cd48a